dart/README.md

```diff
@@ -0,0 +1,218 @@
+import 'package:fury_core/fury_core.dart';
+
+part 'enum_foo.g.dart';
+
+@furyEnum
+enum EnumFoo {
+  A,
+  B
+}
+```
+
+Registration is similar to classes:
+
+```dart
+fury.register($EnumFoo, "example.EnumFoo");
+```
+
+## Type Support
+
+Fury Dart currently supports the following type mappings in XLANG mode:
+
+| Fury Type                  | Dart Type                                       |
+|----------------------------|------------------------------------------------|
+| bool                       | bool                                            |
+| int8                       | fury.Int8                                       |
+| int16                      | fury.Int16                                      |
+| int32                      | fury.Int32                                      |
+| var_int32                  | fury.Int32                                      |
+| int64                      | int                                             |
+| var_int64                  | int                                             |
+| sli_int64                  | int                                             |
+| float32                    | fury.Float32                                    |
+| float64                    | double                                          |
+| string                     | String                                          |
+| enum                       | Enum                                            |
+| named_enum                 | Enum                                            |
+| named_struct               | class                                           |
+| list                       | List                                            |
+| set                        | Set (LinkedHashSet, HashSet, SplayTreeSet)      |
+| map                        | Map (LinkedHashMap, HashMap, SplayTreeMap)      |
+| timestamp                  | fury.TimeStamp                                  |
+| local_date                 | fury.LocalDate                                  |
+| binary                     | Uint8List                                       |
+| bool_array                 | BoolList                                        |
+| int8_array                 | Int8List                                        |
+| int16_array                | Int16List                                       |
+| int32_array                | Int32List                                       |
+| int64_array                | Int64List                                       |
+| float32_array              | Float32List                                     |
+| float64_array              | Float64List                                     |
+
+## Project Structure
+
+The implementation is organized into three main components:
+
+1. **SourceGen**: Located at `dart-fury/packages/fury-core/lib/src/code_gen`
+   Handles static code generation for serialization/deserialization.
+
+2. **FuryCore**: Located at `dart-fury/packages/fury-core/lib/src`
```

Comment:
`fury-core` is the naming style in java, is this an idiom in dart?

Reply:
I'm not familiar about dart, will the strcuture like `lib/fury/core` be better?

Reply:
I mean we could rename `fury-core` to `fury`, and keep `fury-test` as it is. This is the flutter project structure:
![image](https://github.com/user-attachments/assets/6cf7a789-9acc-4189-9402-f5103af9ce0b)